src/ffi/index.ts

```diff
@@ -1,19 +1,159 @@
-import bindings = require('bindings');
+import path from 'node:path';
+import bindings = require('node-gyp-build');
 import logger, { DEFAULT_LOG_LEVEL } from '../logger';
 import { LogLevel } from '../logger/types';
 import { Ffi } from './types';
 
-const ffiLib: Ffi = bindings('pact.node');
-
 export const PACT_FFI_VERSION = '0.4.0';
 
+// supported prebuilds
+// darwin-arm64
+// darwin-x64
+// linux-arm64
+// linux-x64
+// win32-x64
+
+const supportedPlatforms = [
+  'darwin-arm64',
+  'darwin-x64',
+  'linux-arm64',
+  'linux-x64',
+  'win32-x64',
+];
+const platform = `${process.platform}-${process.arch}`;
+
+const supportedPlatformsMessage = [
+  'Supported platforms are: ',
+  ` - ${supportedPlatforms.join('\n - ')}`,
+].join('\n');
+const detectedMessage = `We detected your platform as: \n\n - ${platform}\n`;
+logger.info(detectedMessage);
+if (
+  !supportedPlatforms.includes(platform) &&
+  process.env['UNSUPPORTED_PLATFORM'] !== platform
```

Comment:
I can't see where `UNSUPPORTED_PLATFORM` is actually used, outside of notices?

Reply:
It was, but removed, I tried switching to natively looking up the bindings rather than using another package, which would give us finer grained control over the platform/arch combos, but you have to deal with modules with being loaded sync or async and it was a mission, sooo, they can just raise an issue and we can build ffi bindings in pact-reference if possible :)